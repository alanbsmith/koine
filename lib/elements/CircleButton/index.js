import styled from 'styled-components';
import { rgba } from 'polished';
import { applyStyleModifiers } from 'styled-components-modifiers';

import bgColors from '../../modifiers/bgColors';
import theme from '../../styles/theme';

const CircleButton = styled.button`
  background-color: ${theme.colors.base.chrome000};
  border-radius: 50%;
  border-color: transparent;
  box-shadow: 0 2px 12px 0 ${rgba(theme.colors.base.chrome600, 0.40)};
  cursor: pointer;
  height: 48px;
  outline: none;
  transition: all 0.2s ease;
  width: 48px;

  &:hover {
    transform: scale(1.05);
  }

  ${applyStyleModifiers(bgColors)}
`;

export default CircleButton;
