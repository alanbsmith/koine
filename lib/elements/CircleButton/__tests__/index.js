import React from 'react';
import CircleButton from '../index';
import renderer from 'react-test-renderer';


describe('A CircleButton Element', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <CircleButton />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('applies bgColor modifier', () => {
    const tree = renderer.create(
      <CircleButton modifiers={['bgBlue']} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
