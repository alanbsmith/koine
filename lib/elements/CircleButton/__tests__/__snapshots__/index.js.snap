// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A CircleButton Element applies bgColor modifier 1`] = `
<button
  className="sc-bdVaJa ijtmqC"
/>
`;

exports[`A CircleButton Element renders correctly 1`] = `
<button
  className="sc-bdVaJa eTpEPC"
/>
`;
